### PR TITLE
Add options support; Pass arguments to setup binstubs.

### PR DESCRIPTION
**Pull request by johnallen3d**
~~In addition to supporting this new option (minor)~~ (change removed) this commit adds support for defining options in configuration and having them rendered in help.

* New based on comments below:

The additional arguments to the setup command are now passed through to each of the setup scripts.

### CODE REVIEW

File: bin/_common [hunk 1]
```diff
@@ -11,6 +11,10 @@ commands_json() {
   cat /usr/local/bin/config/commands.json
 }
 
+command_json() {
```
Comment (johnallen3d) [inline bin/_common]: This concept was being defined repeatedly. Refactored it into a function.

File: bin/_help [hunk 2]
```diff
@@ -7,34 +7,33 @@
 source "/usr/local/bin/_common"
 
 print_common_usage() {
-  echo "Usage: nib ${1} [SERVICE]"
+  echo "Usage: nib $COMMAND [SERVICE]"
```
Comment (johnallen3d) [inline bin/_help]: Magic variable `$COMMAND` is always set, no need to pass/eval argument in this method. Should this not be part of this PR?

File: bin/_help [hunk 3]
```diff
@@ -7,34 +7,33 @@
...
 }
 
 print_help_options() {
-  echo "Options:
-  -h, --help      Print usage
-      --version   Print version information
-"
```
Comment (johnallen3d) [inline bin/_help]: This prior implementation was only useful for `nib --help`. This has been moved there making this function more flexible.

File: bin/setup [hunk 4]
```diff
@@ -17,7 +26,10 @@ SCRIPT="
   else
     gem install bundler
     bundle install --jobs 4
-    rake db:create db:migrate
+
+    if [ $skip_db_setup = false ]; then
```
Comment (johnallen3d) [inline bin/setup]: This is the actual functionality change.

**Comment by johnallen3d** [comment]
👀  @jackross @blankenshipz

**Comment by jackross** [comment]
@johnallen3d So what you're saying is that the regular use case would be for `nib setup` to run `rake db:create db:migrate`, because that's what most users would want, with which I think I agree.  However, if the project is not a `rails` or database-based project, I would be annoyed to have to `skip_db_setup` every time.  Can we check to see somehow if `rake db:create` is an available command in the project, and skip it by default if it is not available?

**Comment by johnallen3d** [comment]
Yes, the working assumption was that the most common usage would be for Rails projects so most users would be fine with the default. Of course newer Rails projects already define `./bin/setup` in which case the defaults would not apply...

Do you think the idea of having a default is too fraught with peril? If not, then yes, I like the idea of adding a check to see if `rake db:create` is available or even `rake` itself.

**Comment by jackross** [comment]
I forgot that running `.bin/setup` if it existed would be the default.  That would make me lean towards NOT running the rake tasks by default.  However, what if we could respect a `NIB_SETUP_SKIP_DB_SETUP` environment variable, added in `docker-compose.yml` or in a `.env.nib` file?

**Comment by blankenshipz** [comment]
I think I lean towards removing the db stuff from the default behavior, but I can see the case for the flag. It brings up another interesting question - we should make sure that any additional args passed to 'nib setup' are passed to any user defined bin/setup I can see that as a potentially useful feature

**Comment by johnallen3d** [comment]
Ok, I think we're all in agreement that the rake db commands should just be removed which would negate the need for the `--skip-db-setup` argument. Do we think that having the option to support arguments in general is valid? If so I can change the focus of this PR.

Good point about additional argument @blankenshipz! Should those arguments be passed along to the before/after hooks as well? Do we see actual value in the before/after hooks or is that a premature optimization?

**Comment by blankenshipz** [comment]
I still think the hooks are a good convention, we may find some more things to add to the default setup in the future. I'm not sure what you mean when you say

>Do we think that having the option to support arguments in general is valid? If so I can change the focus of this PR.

Are you talking about the `options` array in the `commands.json`?

**Comment by johnallen3d** [comment]
Ok, keeping the hooks is reasonable. The question remains, should the additional arguments you mentioned (quote below) be passed to each script or just the primary `./bin/setup`?

> we should make sure that any additional args passed to 'nib setup' are passed to any user defined bin/setup I can see that as a potentially useful feature

Yes, when I said `the option to support arguments in general` I was referring to the `options` array in `commands.json` and the other changes in this PR to support them. I ask because `--skip-db-setup` is the only currently supported option.

**Comment by blankenshipz** [comment]
I think that any args should be passed to whichever script is selected, whose to say that the user hasn't defined their own `--skip-db--setup`

**Comment by johnallen3d** [comment]
@blankenshipz - I've updated the Title, Description and code. Can you take another look at this?

**Comment by blankenshipz** [comment]
LGTM!
